vscode-ext/src/StatusBarComponent.ts

```diff
@@ -38,36 +48,156 @@ export class StatusBarComponent {
+                    this.view.text = "$(debug-start) Et tu, Brute?";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipMenuText = stoppedTooltipMenu;
                     break;
                 case ServerStatus.FailedToStart:
                     this.view.text = "$(warning) Failed to start Caesar";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipMenuText = stoppedTooltipMenu;
                     break;
                 case ServerStatus.Starting:
-                    this.view.text = "$(sync~spin) Starting Caesar...";
+                    this.view.text = "$(loading~spin) Starting Caesar...";
+                    this.view.command = "caesar.showOutput"
                     break;
                 case ServerStatus.Ready:
-                    this.view.text = "$(check) Caesar Ready";
+                    this.handleReadyStatus();
+                    this.view.command = "caesar.showOutput";
+                    this.tooltipMenuText = runningTooltipMenu;
                     break;
                 case ServerStatus.Verifying:
                     this.view.text = "$(sync~spin) Verifying...";
-                    break;
-                case ServerStatus.Finished:
-                    this.view.text = "$(check) Verified";
+                    this.view.command = "caesar.showOutput"
                     break;
             }
+
+            this.renderTooltip();
             this.view.show();
         } else {
             this.view.hide();
         }
     }
+
+    private renderTooltip() {
+        if (this.tooltipStatusText.value === "") {
+            this.view.tooltip = this.tooltipMenuText;
+            this.view.tooltip.isTrusted = true;
+            return;
+        }
+        this.view.tooltip = new vscode.MarkdownString(this.tooltipMenuText.value + "\n\n --- \n" + this.tooltipStatusText.value, true);
+        this.view.tooltip.isTrusted = true;
+    }
+
+
+    private handleReadyStatus() {
+        let tooltipString = new vscode.MarkdownString("", true);
+
+        let someError = false;
+        let someVerified = false;
+
+        const editorsWithResults = vscode.window.visibleTextEditors.filter((editor) => {
+            return editor.document.languageId === "heyvl" // Only HeyVL files
+                && (this.verifyStatus.get({ uri: editor.document.uri.toString() }) ?? []).length !== 0; // And only files with results
+        });
+
+        if (editorsWithResults.length === 0) {
+            // No HeyVL files open or no results yet
+            this.view.text = "$(thumbsup-filled) Caesar Ready";
+            return;
+        }
+
+        for (const editor of editorsWithResults) {
+            const document_id: TextDocumentIdentifier = { uri: editor.document.uri.toString() };
+
+            // The default will not be used because the filter above ensures that the document is in the map.
```

Comment:
Why handle the default case then with `?? []`? Why not just throw an error then?